autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -17,7 +17,7 @@
 if TYPE_CHECKING:
     from backend.server.rest_api import AgentServer
 
-from backend.data import db
+from backend.data import db, redis
```

Comment:
Basically, this is the main changes, locking use redis, and remove the rest_server.lock